app/userapps/handlers.js

```diff
@@ -7,6 +7,10 @@ exports.create = (req, res, next) => {
 
   App.get(app_id)
     .then(app => app.grant_user_access(user_id, 'readonly', true))
+    .then(() => User.get(req.user.auth0_id))
+    .then((user) => {
+      req.session.passport.user.userapps = user.data.userapps;
+    })
```

Comment:
Since this has been repeated a few times, it would be worth extracting into a function:
```javascript
function update_current_user(req) {
  return User.get(req.user.auth0_id)
    .then((user) => {
      req.session.passport.user.userapps = user.data.userapps;
    });
}

// and then ...
App.get(app_id)
  .then(app => app.grant_user_access(...))
  .then(update_current_user(req))
  .then(() => {
    res.redirect(...);
  })
```